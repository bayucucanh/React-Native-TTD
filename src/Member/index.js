import { View, Text } from 'react-native'
import React from 'react'

const Member = () => {
  const members = [{id: 1, name: 'maman'}, {id: 2, name: 'jajang'}];
  return (
    <View>
      <Text>Member</Text>
      {members.map(member => (
        <Text key={member.id} testID='memberDetail'>{member.name}</Text>
      ))}
    </View>
  )
}
export default Member