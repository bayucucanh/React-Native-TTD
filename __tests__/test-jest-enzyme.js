import React from "react";
import { shallow, configure } from "enzyme";
import toJson from "enzyme-to-json";
import Adapter from 'enzyme-adapter-react-16'
configure({adapter: new Adapter(), disableLifecycleMethods: true})

import Member from "../src/Member";

const members = [{id: 1, name: 'maman'}, {id: 2, name: 'jajang'}];

describe('Member Component', () => { 
  it('should render without issues', () => {
    const component = shallow(
      <Member members={members} />
    );

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot()
  })

  it('should render all item in members', () => {
    const wrapper = shallow(
      <Member members={members} />
    );

    expect(wrapper.find({testID: 'memberDetail'}).length).toBe(2)
  })

  it('should render render correct names', () => {
    const wrapper = shallow(
      <Member members={members} />
    );

    expect(wrapper.find({testID: 'memberDetail'}).forEach((node, index) => {
      expect(node.props().children).toBe(members[index].name)
    }))
  })
 })