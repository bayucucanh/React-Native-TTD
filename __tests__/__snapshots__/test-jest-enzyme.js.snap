// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Member Component should render without issues 1`] = `
<View>
  <Text>
    Member
  </Text>
  <Text
    key="1"
    testID="memberDetail"
  >
    maman
  </Text>
  <Text
    key="2"
    testID="memberDetail"
  >
    jajang
  </Text>
</View>
`;
